api/tests/tooling/server/http-test-server.js

```diff
@@ -37,9 +37,11 @@ class HttpTestServer {
     });
 
     if (enableAuthentication) {
-      this.hapiServer.auth.scheme('jwt-scheme', security.scheme);
-      this.hapiServer.auth.strategy('default', 'jwt-scheme');
-      this.hapiServer.auth.default('default');
+      this.hapiServer.auth.scheme(authentication.schemeName, authentication.scheme);
+      authentication.strategies.map((strategy) => {
+        this.hapiServer.auth.strategy(strategy.name, authentication.schemeName, strategy.configuration);
+      });
+      this.hapiServer.auth.default(authentication.defaultStrategy);
```

Comment:
**question**
J'ai pas trouvé de test qui permet de vérifier qu'on utilise bien la stratégie par défaut par défaut si on ne précise la stratégie en entrée. Est-ce que ça serait testable ? Avec un stub et expect withArgs peut être ?

Reply:
C'est bien testé, car la route n'a pas d'option `auth`, donc elle utilise la stratégie par défaut
```
{
  method: 'GET',
  path: '/foo',
  config: {
    handler: () => {
      return 'bar';
    },
  }
```

Mais le titre du cas de test n'est pas explicite
Je l'ai renommé en `when authentication is enabled, using default strategy`